### PR TITLE
update Roslyn references

### PR DESCRIPTION
**Pull request by brettfo**
This includes updating the document highlights API exposed by Roslyn.

NOTE: this PR is initially expected to fail as I have had to include a dummy package version number.  I'll update this PR with the real package number once it's available.

@KevinRansom @Pilchie @cyrusnajmabadi

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs [hunk 1]
```diff
@@ -45,7 +45,7 @@ type internal FSharpCompletionProvider
         |> List.filter (fun (keyword, _) -> not (PrettyNaming.IsOperatorName keyword))
         |> List.sortBy (fun (keyword, _) -> keyword)
         |> List.mapi (fun n (keyword, description) ->
-             CommonCompletionItem.Create(keyword, Nullable Glyph.Keyword, sortText = sprintf "%06d" (1000000 + n))
+             CommonCompletionItem.Create(keyword, CompletionItemRuls.Default, Nullable Glyph.Keyword, sortText = sprintf "%06d" (1000000 + n))
```
Comment (CyrusNajmabadi) [inline vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs]: Why did you change this?

Comment (CyrusNajmabadi) [inline vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs]: Note: CompletionItemRuls is mispelled.

Comment (brettfo) [inline vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs]: This is a port from your last PR (last week?) where I had to make this change.  Is this change not in your latest PR dotnet/roslyn#19525?

Comment (brettfo) [inline vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs]: I never said I could spell :smile:

Once we have an official package I can build against I'll do a local build to catch anything outstanding.

File: vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs [hunk 2]
```diff
@@ -133,7 +133,7 @@ type internal FSharpFindUsagesService
                         | _ ->
                             for referenceDocSpan in referenceDocSpans do
                                 for definitionItem in definitionItems do
-                                    let referenceItem = SourceReferenceItem(definitionItem, referenceDocSpan)
+                                    let referenceItem = SourceReferenceItem(definitionItem, referenceDocSpan, true)
```
Comment (CyrusNajmabadi) [inline vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs]: Note: you may want to comment that you're just passing 'true' for now until you get info on if this is actually a write or not from your compiler.

**Comment by Pilchie** [comment]
What are you meaning by "official" packages?  I would expect signed packages to be published to myget as part of the signed build that Roslyn is going to insert.

**Comment by brettfo** [comment]
@Pilchie that's what I meant; a signed package on MyGet that I can reference.

File: vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs [hunk 3]
```diff
@@ -133,7 +133,7 @@ type internal FSharpFindUsagesService
                         | _ ->
                             for referenceDocSpan in referenceDocSpans do
                                 for definitionItem in definitionItems do
-                                    let referenceItem = SourceReferenceItem(definitionItem, referenceDocSpan)
+                                    let referenceItem = SourceReferenceItem(definitionItem, referenceDocSpan, true) // defaulting to `true` until we can officially determine if this usage is a write
```
Comment (saul) [inline vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs]: Can we put an issue in about this?

**Comment by Pilchie** [comment]
@brettfo I think https://github.com/dotnet/roslyn/pull/19525 being done in a non-breaking way means this isn't urgent to get from MyGet, but we do still need to address the completion and Tokenizer issues right?

**Comment by brettfo** [comment]
Correct.  As soon as packages exist I'll push this through anyways, just to have it done.  It'll also fix an issue with our glyphs in the completion list since the backing enum has changed.

**Comment by brettfo** [comment]
@dotnet-bot test vsi please // this should pick up recent VSI test fixes

**Comment by KevinRansom** [comment]
@dotnet-bot test this please
